### PR TITLE
Modularize Inference script

### PR DESCRIPTION
**Pull request by uSaiPrashanth**
[Inference script](https://github.com/EleutherAI/variance-across-time/blob/sai/inference.py) calculates logits across all models from the given dataset (a subset of train, test and out_of_distribution). 

Further metric calculations, like Jenson shannon divergance are handled by this [pipeline](https://github.com/EleutherAI/variance-across-time/blob/sai/inference_stats/base.py). [see example of jsd here](https://github.com/EleutherAI/variance-across-time/blob/sai/inference_stats/jenson_shannon_divergance.py)

### CODE REVIEW

**Review by norabelrose — APPROVED** [review]
lgtm
